project4/name_server/nameserver.cpp

```diff
@@ -5,14 +5,12 @@ Nameserver::Nameserver(const std::filesystem::path config_file) {
   client_threadpool_ = std::make_shared<thread_pool::ThreadPool>();
 }
 
-bool Nameserver::Enter() {
-  return false;
-}
-void Nameserver::Exit() {
+bool Nameserver::Enter() { return false; }
+
+void Nameserver::Exit() {}
 
-}
 void Nameserver::HandleRequest(
```

Comment:
What's the point of a `HandleRequest` overload for a generic protobuf message? Since `NameServerMessage` is the container message, I'd think that this was right before.

Reply:
The point of genericizing was to be able to override it in `Bootstrap` to handle different requests accordingly, because it can be a `NameServerMessage` for a nameserver, or a `BootstrapMessage` for a bootstrap. The `Bootstrap` would call `Nameserver::HandleRequest()` to handle anything else that isn't a `BootstrapMessage`.